behind-the-veil-siteroot/imports/ui/components/pages/register/CreateAccountPage.jsx

```diff
@@ -18,97 +18,223 @@ import UrlBasePath from "../../../enums/UrlBasePath";
-                console.log('User created successfully!');
-                console.log(Meteor.userId());
-                console.log(Meteor.user());
-                Meteor.logout()
-                // After successful activation, navigate to activation completed page
-                navigate(`/${UrlBasePath.REGISTER}/activateAccount?username=${username}`);
-            }
-        });
+                } else {
+                    // verify its email right after it is created
+                    Meteor.call("verify_email", Meteor.userId());
+                    Meteor.logout()
+
+                    // After successful activation, navigate to activation completed page
+                    navigate(`/${UrlBasePath.REGISTER}/accountCreated`);
+                }
+            });
+        }
     };
 
-    const TextInput = ({labelText, id, name, placeholder, type = 'text', autoComplete = 'off'}) => {
-        const [value, setValue] = useState('');
+    const TextInput = ({labelText, id, name, error, placeholder, type = 'text', autoComplete = 'off'}) => {
+        const existingFormValue = document.getElementById(id) ? document.getElementById(id).value : '';
+        const [value, setValue] = useState(existingFormValue);
 
         const handleChange = (e) => {
-            const inputValue = e.target.value;
+            const inputValue = e.target.value.trim();
             setValue(inputValue);
         };
 
         return (
-            <Input
-                label={<label htmlFor={id} className="main-text">{labelText}</label>}
-                type={type}
-                id={id}
-                name={name}
-                placeholder={placeholder}
-                autoComplete={autoComplete}
-                value={value}
-                onChange={handleChange}
-                className={"w-full"}
-            />
+            <div className="flex flex-col gap-1">
+                <Input
+                    label={<label htmlFor={id} className="main-text">{labelText}</label>}
+                    type={type}
+                    id={id}
+                    name={name}
+                    placeholder={placeholder}
+                    autoComplete={autoComplete}
+                    value={value}
+                    onChange={handleChange}
+                    className={"w-full"}
+                />
+                {error && <span className="text-cancelled-colour">{error}</span>}
+            </div>
         );
     };
 
+    const RadioInput = ({text, id}) => {
+        // handle change
+        const handleChange = (e) => {
+            const inputValue = e.target.value;
+            setAccountType(inputValue);
+        };
+
+        // set default checked one
+        if (id === accountType) {
+            return (
+                <div className="flex flex-col items-center justify-center gap-1">
+                    <label className="main-text text-dark-grey" htmlFor="artist">{text}</label>
```

Comment:
Is the htmlFor prop supposed to point to the id of the input instead of being hardcoded to artist? Both inputs here have html="artist"

Reply:
good catch!